core/dbt/context/providers.py

```diff
@@ -675,10 +674,8 @@ def packages_for_node(self) -> Iterable[Project]:
         package_name = self._node.package_name
 
         if package_name != self._config.project_name:
-            if package_name not in dependencies:
-                # I don't think this is actually reachable
-                raise PackageNotInDepsError(package_name, node=self._node)
-            yield dependencies[package_name]
+            if package_name in dependencies:
+                yield dependencies[package_name]
```

Comment:
For posterity, thinking out loud about the implications of this change: 
* This method is used to load the vars available for a particular jinja (model-scoped) context via the `vars` macro.
* This method falls back on the running project's vars if the node has no associated packages. If it returns the wrong package -- this could lead to a 'no var found' issue at runtime.
* Before external models, it was not possible to have a model node exist in the project without a corresponding package, which is probably why this code path was not reachable
* With external, this is totally possible now -- so explicitly supporting both paths feels totally reasonable.